src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -235,6 +239,31 @@ public void writeLogTo(XMLOutput out) throws IOException {
         @Inject
         private transient GitHubHookRegisterProblemMonitor monitor;
 
+        @Inject
+        private transient SCMTrigger.DescriptorImpl scmTrigger;
+
+        private transient int maximumThreads = Integer.MIN_VALUE;
+
+        public DescriptorImpl() {
+            checkThreadPoolSize();
+        }
+
+        /**
+         * Update the {@link java.util.concurrent.ExecutorService} instance.
+         */
+        /*package*/
+        synchronized void checkThreadPoolSize() {
+            if (scmTrigger != null) {
+                int count = scmTrigger.getPollingThreadCount();
+                if (maximumThreads != count) {
```

Comment:
Because we want to catch when the thread pool size is changed in any way (up or down)